lib/pf/Authentication/Source/PaypalSource.pm

```diff
@@ -0,0 +1,322 @@
+package pf::Authentication::Source::PaypalSource;
+
+=head1 NAME
+
+pf::Authentication::Source::PaypalSource
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::Authentication::Source::PaypalSource
+
+=cut
+
+use strict;
+use warnings;
+use Moose;
+use pf::config qw($FALSE $TRUE $default_pid $fqdn);
+use pf::Authentication::constants;
+use pf::util;
+use pf::log;
+use WWW::Curl::Easy;
+use JSON::XS;
+use List::Util qw(first pairmap);
+use URI::Escape qw(uri_escape uri_unescape);
+use HTTP::Status qw(is_success :constants);
+use IPC::Open2;
+
+our $PAYPAL_SANDBOX_URL = 'https://www.sandbox.paypal.com/cgi-bin/webscr';
+our $PAYPAL_URL = 'https://www.paypal.com/cgi-bin/webscr';
```

Comment:
In my opinion this is much more a constant.

This is not something that the user should be changing.

Should the URL for this change, a patch on our side should be created.

The issue with putting it in the configuration is, if the URL changes and we patch it in the package, the existing sources are not upgraded.

A perfect example of that is the scopes of the OAuth sources that constantly evolve but that aren't upgraded when we patch it because it doesn't affect existing sources...